### PR TITLE
Expose stateful header sets and make the check case-insensitive

### PR DESCRIPTION
**Pull request by nyaxt**
HTTP Header names are case-insensitive.

### CODE REVIEW

File: go/signedexchange/signedexchange_test.go [hunk 1]
```diff
@@ -240,4 +240,15 @@ func TestSignedExchangeStatefulHeader(t *testing.T) {
...
 	}
+
+	// Header names are case-insensitive.
+	u, _ = url.Parse("https://example.com/")
+	header = http.Header{}
+	header.Add("cOnTent-TyPe", "text/html; charset=utf-8")
+	header.Add("setProfile", "profile X")
+
+	_, err = NewExchange(u, nil, 200, header, []byte(payload), 16)
+	if err == nil {
```
Comment (hajimehoshi) [inline go/signedexchange/signedexchange_test.go]: `if _, err = NewExchange(...); err == nil {`

File: go/signedexchange/signedexchange.go [hunk 2]
```diff
@@ -31,36 +31,38 @@ var HeaderMagicBytes = []byte("sxg1-b1\x00")
...
+		"authentication-info":       struct{}{},
+		"optional-www-authenticate": struct{}{},
+		"proxy-authenticate":        struct{}{},
+		"proxy-authentication-info": struct{}{},
+		"sec-webSocket-accept":      struct{}{},
+		"set-cookie":                struct{}{},
+		"set-cookie2":               struct{}{},
+		"setprofile":                struct{}{},
+		"www-authenticate":          struct{}{},
 	}
```
Comment (hajimehoshi) [inline go/signedexchange/signedexchange.go]: As we discussed offline, let's have functions instead of exposing variables.

**Comment by nyaxt** [comment]
PTAL

File: go/signedexchange/signedexchange.go [hunk 3]
```diff
@@ -29,38 +29,15 @@ type Exchange struct {
...
-		"SetProfile":                struct{}{},
-		"WWW-Authenticate":          struct{}{},
-	}
-)
-
 func NewExchange(uri *url.URL, requestHeaders http.Header, status int, responseHeaders http.Header, payload []byte, miRecordSize int) (*Exchange, error) {
-	for h, _ := range statefulRequestHeaders {
-		if _, ok := requestHeaders[h]; ok {
-			return nil, fmt.Errorf("signedexchange: stateful request header %q can't be captured inside signed exchange", h)
+	for name, _ := range requestHeaders {
```
Comment (hajimehoshi) [inline go/signedexchange/signedexchange.go]: `for name := range requestHeaders {`

File: go/signedexchange/signedexchange.go [hunk 4]
```diff
@@ -29,38 +29,15 @@ type Exchange struct {
...
+		if IsStatefulRequestHeader(name) {
+			return nil, fmt.Errorf("signedexchange: stateful request header %q can't be captured inside signed exchange", name)
 		}
 	}
-	for h, _ := range statefulResponseHeaders {
-		if _, ok := responseHeaders[h]; ok {
-			return nil, fmt.Errorf("signedexchange: stateful response header %q can't be captured inside signed exchange", h)
+	for name, _ := range responseHeaders {
```
Comment (hajimehoshi) [inline go/signedexchange/signedexchange.go]: ditto

File: go/signedexchange/signedexchange.go (hunk 3 above)
Comment (nyaxt) [reply go/signedexchange/signedexchange.go]: done

File: go/signedexchange/signedexchange.go (hunk 4 above)
Comment (nyaxt) [reply go/signedexchange/signedexchange.go]: done

**Pull request merged**